hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -944,4 +944,12 @@
      */
     CacheStatistics getLocalCacheStatistics();
 
+    /**
+     * Opens cache if available (not destroyed).
+     * If cache is already open, does nothing.
+     *
+     * @throws java.lang.IllegalStateException if cache is already destroyed
+     */
+    void open();
```

Comment:
No, `ICache` should not extend internal interfaces. Only proxies can implement/extend internal interfaces/classes. You can make `ClientProxy`/`ClientCacheProxy` extend/implement internal one.